### PR TITLE
V2 - Documentation pages

### LINKED ISSUE #323: V2 - Docs
Setting up this issue to track requirements as I am working on the documentation theme/layout for V2.

- [ ] Layout
- [ ] Side navigation
- [ ] Styling
- [ ] ...

### PR DESCRIPTION
**Pull request by abuuzayr**
Closes #323

### CODE REVIEW

**Comment by flybayer** [comment]
Nice work!

Deploy is failing:
image

**Comment by abuuzayr** [comment]
> Nice work!
> 
> Deploy is failing:
> image

thanks for the heads up! let me sort that out and try a build before pushing for review.

**Comment by abuuzayr** [comment]
There are still some parts that are incomplete, but the basic build is ok. 

1. Mobile menu isn't working for docs page
2. Sidebar should become a drop down as per the design in mobile view
3. Dark style for docs is not appearing - this is fine on local build

should I open for review or fix any/all of these first? Just in case this PR is a blocker for other issues..

File: new-site/app/layouts/SidebarLayout.js [hunk 1]
```diff
@@ -49,41 +42,35 @@ function Nav({ nav, children, fallbackHref }) {
...
-                      "px-3 mb-3 lg:mb-3 uppercase tracking-wide font-semibold text-sm lg:text-xs",
-                      {
-                        "text-gray-900": publishedItems.length > 0,
-                        "text-gray-400": publishedItems.length === 0,
-                      }
-                    )}
-                  >
-                    {category}
-                  </h5>
+                <li key={category.title} className="my-5">
```
Comment (mikeesto) [inline new-site/app/layouts/SidebarLayout.js:55]:
```suggestion
                <li key={category.title.props.title} className="my-5">
```
Fixes _encountered two children with same key_ warning

File: new-site/app/layouts/SidebarLayout.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
-import Link from "next/link"
-import { useRouter } from "next/router"
+import { Link, useRouter } from "blitz"
 import { createContext, forwardRef, useRef } from "react"
```
Comment (mikeesto) [inline new-site/app/layouts/SidebarLayout.js:2]:
```suggestion
import { createContext, forwardRef, useRef, Fragment } from "react"
```

File: new-site/app/layouts/SidebarLayout.js [hunk 3]
```diff
@@ -49,41 +42,35 @@ function Nav({ nav, children, fallbackHref }) {
...
-                        href={item.href}
-                        isActive={item.href === router.pathname}
-                        ref={item.href === router.pathname ? activeItemRef : undefined}
-                        isPublished={item.published !== false}
-                        fallbackHref={fallbackHref}
-                      >
-                        {item.shortTitle || item.title}
-                      </NavItem>
+                    {(fallbackHref ? category.pages : publishedItems).map((item, i) => (
+                      <>
```
Comment (mikeesto) [inline new-site/app/layouts/SidebarLayout.js:59]:
```suggestion
                      <Fragment key={i}>
```
Key needs to be on fragment

File: new-site/app/layouts/SidebarLayout.js [hunk 4]
```diff
@@ -49,41 +42,35 @@ function Nav({ nav, children, fallbackHref }) {
...
+                        <NavItem
+                          key={i}
```
Comment (mikeesto) [inline new-site/app/layouts/SidebarLayout.js:61]:
```suggestion
```
Remove

File: new-site/app/layouts/SidebarLayout.js [hunk 5]
```diff
@@ -49,41 +42,35 @@ function Nav({ nav, children, fallbackHref }) {
...
+                          ref={item.href === router.pathname ? activeItemRef : undefined}
+                          isPublished={item.published !== false}
+                          fallbackHref={fallbackHref}
+                        >
+                          {item.shortTitle || item.title}
+                        </NavItem>
+                        {item.href === router.pathname && toc && toc.length ? (
+                          <TableOfContents tableOfContents={toc} />
+                        ) : null}
+                      </>
```
Comment (mikeesto) [inline new-site/app/layouts/SidebarLayout.js:73]:
```suggestion
                      </Fragment>
```

File: new-site/app/layouts/ContentsLayout.js [hunk 6]
```diff
@@ -4,10 +4,11 @@ import Link from "next/link"
 import { SidebarLayout, SidebarContext } from "@/layouts/SidebarLayout"
 import { PageHeader } from "@/components/PageHeader"
 import clsx from "clsx"
+import { ReactComponent as ArrowIcon } from "@/img/icons/nav-arrow.svg"
 
 export const ContentsContext = createContext()
 
-function TableOfContents({ tableOfContents, currentSection }) {
+export function TableOfContents({ tableOfContents, currentSection }) {
   let sidebarContext = useContext(SidebarContext)
```
Comment (mikeesto) [inline new-site/app/layouts/ContentsLayout.js:12]: `setNavIsOpen` doesn't seem to be defined in SidebarContext

To reproduce error go to: `/docs/getting-started#quick-start`

File: new-site/app/pages/docs/how-the-community-operates.mdx [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+</table>
+
+## L2 Maintainers
+
+Level 2 maintainers are the backbone of the community and are responsible for reviewing and merging PRs. They are watchdogs over the code, ensuring code quality, correctness, and security. They also help facilitate a rapid pace of progress.
+
+For more details, see [Being an L2 Maintainer](./maintainers)
+
+<table>
+  <tr>
```
Comment (mikeesto) [inline new-site/app/pages/docs/how-the-community-operates.mdx:116]:
```suggestion
  <tbody>
    <tr>
```
Add tbody here to solve DOM warning

File: new-site/app/pages/docs/how-the-community-operates.mdx [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+        <img src="https://avatars1.githubusercontent.com/u/14912729?v=4" width="100px;" alt="" />
+        <br />
+        <sub>
+          <b>Simon Knott</b>
+        </sub>
+      </a>
+      <br />
+      SuperJSON
+    </td>
+  </tr>
```
Comment (mikeesto) [inline new-site/app/pages/docs/how-the-community-operates.mdx:150]:
```suggestion
  </tr>
</tbody>
```

**Review by mikeesto — COMMENTED** [review]
Awesome work. Made a couple of suggestions to solve some of the warnings that were spamming up the console.

The dark styling is working well locally for me too - not sure what's happening on deployment

File: new-site/app/layouts/ContentsLayout.js (hunk 6 above)
Comment (abuuzayr) [reply new-site/app/layouts/ContentsLayout.js:12]: OK let me have a look at it

**Comment by abuuzayr** [comment]
Thank you so much @mikeesto for having a look at this and testing it out!

**Comment by flybayer** [comment]
@abuuzayr thank you so much your hard work on this!!!

I recorded a video of feedback for you: https://www.loom.com/share/9f17bcb704ce4dc699293782dfcf6abe

**Comment by abuuzayr** [comment]
thanks a lot @flybayer for the video and review! am working on the changes and will let you know when it is ready

**Comment by abuuzayr** [comment]
@flybayer you mentioned the icons were missing for some sections like Community, CLI, Auth, Recipes, etc. But I am unable to find them in Figma. Perhaps I am looking in the wrong place, could you show me where I can find them?

**Comment by flybayer** [comment]
@abuuzayr ah, yes, sorry about that. Ignore those for now, I'll ask the designer.

**Comment by flybayer** [comment]
Hey @abuuzayr, FYI I'm wanting to get the site fully finished asap this week for the beta launch, likely next week.

**Comment by abuuzayr** [comment]
@flybayer absolutely, sorry about that. will have more time tomorrow and the day after. will put a priority on it!

**Comment by flybayer** [comment]
Also @abuuzayr how did you copy all the docs to this new setup? Was there a script you ran or something?

After this is merged, I will merge into canary branch and then we need to update all the docs from the canary docs.

**Comment by abuuzayr** [comment]
@flybayer it was a simple copy only which i would manually run, its definitely not in sync with the current docs

**Comment by flybayer** [comment]
@abuuzayr ok cool. Just let me know when you are done with your changes and then I'll merge.

**Comment by flybayer** [comment]
heads up, you have some merge conflicts.

Also fyi, probably on Monday I'll go ahead and merge this and merge to canary branch and then work from that branch for polish for Wednesday's launch

**Comment by abuuzayr** [comment]
Thanks @flybayer! i'm left with the styling for the mobile topics selection, then i'll let you know when its all done. 

so sorry for the delay, had a crazy week - hope you had better one!

**Comment by abuuzayr** [comment]
@flybayer I resolved all the conflicts, and completed the styling for the topics selection. I tried as much as possible to follow the design & guidelines, but there's definitely a lot of places to be polished or places I might have missed, the biggest issue right now being the dark mode prose styles not being generated on deployment (but ok on local).

I think however that it is OK to be merged so you or others can work on polishing it up.

**Comment by flybayer** [comment]
Ok @abuuzayr thank you SO SO much for your help on this!!

**Review by flybayer — APPROVED** [review]

**Pull request merged**